core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -504,9 +542,28 @@ private[spark] object JsonProtocol {
       case `executorRemoved` => executorRemovedFromJson(json)
       case `logStart` => logStartFromJson(json)
       case `metricsUpdate` => executorMetricsUpdateFromJson(json)
+      case `sqlExecutionStart` => sqlExecutionStartFromJson(json)
+      case `sqlExecutionEnd` => sqlExecutionEndFromJson(json)
```

Comment:
This mach really needs `case other:String => throw new IOException(s"Unknown Event $other") message, so that when new events are added in future, the old replay code will at least display a meaningful message. Maybe that's for another pull req though.